session.go

```diff
@@ -762,7 +765,6 @@ func (s *session) Execute(goCtx goctx.Context, sql string) (recordSets []ast.Rec
 			stmt, err := compiler.Compile(goCtx, stmtNode)
 			if err != nil {
 				log.Warnf("[%d] compile error:\n%v\n%s", connID, err, sql)
-				terror.Log(errors.Trace(s.RollbackTxn(goCtx)))
```

Comment:
I think we should add a flag to mark the transaction to be rollback. Then if auto commit is false, the further `rollback` or `commit` should do rollback, else rollback directly.

Reply:
Now the behavior is ... rollback current txn, and continue the remain statements in a new txn.